shared/templates/template_common.py

```diff
@@ -155,30 +150,64 @@ def csv_map(filename, method, skip_comments = True, target = None):
     """
 
     with open(filename, 'r') as csv_file:
-        filtered_file = filter_out_csv_lines(csv_file, target)
+        filtered_file = filter_out_csv_lines(csv_file, language)
 
         csv_lines_content = csv.reader(filtered_file)
 
         try:
-            map(method, csv_lines_content)
+            for csv_line in csv_lines_content:
+                method(language, csv_line)
         except UnknownTargetError as e:
             sys.stderr.write(str(e) + "\n")
-            sys.exit(EXIT_UNKNOWN_TARGET)
+            sys.exit(ExitCodes.UNKNOWN_TARGET)
+
+def parse_args(csv_format):
+    p = argparse.ArgumentParser()
+
+    sp = p.add_subparsers(help="actions")
+
+    make_sp = sp.add_parser('build', help="Build scripts")
+    make_sp.set_defaults(action=ActionType.BUILD)
+
+    input_sp = sp.add_parser('list-inputs', help="Generate input list")
+    input_sp.set_defaults(action=ActionType.INPUT)
+
+    output_sp = sp.add_parser('list-outputs', help="Generate output list")
+    output_sp.set_defaults(action=ActionType.OUTPUT)
+
+    p.add_argument('-s','--shared_dir', action="store", help="Shared directory")
+    p.add_argument('--language', action="store", default=None, required=True,
+                   help="Scripts of which language should we generate?")
+    p.add_argument('-c',"--csv", action="store", required=True,
+                   help="csv filename.\n" + csv_format)
+    p.add_argument('-o','--output_dir', action="store", required=True,
+                   help="output dir")
+    p.add_argument('-i','--input_dir', action="store", required=True,
+                   help="templates dir")
+
+    args, unknown = p.parse_known_args()
+
+    return (args,unknown)
+
+
 
-def main(argv, help_callback, process_line_callback):
+def main(argv, csv_format, process_line_callback):
 
-    argv_len = len(argv)
+    parsed, unknown = parse_args(csv_format)
 
-    if argv_len < 3:
-        help_callback()
-        sys.exit(1)
+    if unknown:
+        sys.stderr.write(
+            "Unknown positional arguments " + ",".join(unknown) + ".\n"
+        )
+        sys.exit(ExitCodes.ERROR)
 
-    target = sys.argv[1]
-    filename = sys.argv[2]
+    global output_dir
+    global action
+    global product_input_dir
```

Comment:
I think, the only way is require modification of every single `create_*py` file, because you need to pass these values into `file_from_template()`.

I could rather to put it into classes and use `self` , but it will produce huge diff.
From your point of view, does it worth to do it now ?